File: fine-line/src/main/kotlin/domain/group/models/Group.kt
package domain.group.models

import domain.fineType.models.FineType
import domain.user.models.User

data class Group(
    val groupId: Int,
    val name: String,
    val admins: List<User>,
    val members: List<User>,
    val pendingMembers: List<User>,
    val fineTypes: List<FineType>
)

File: client/src/lib/models/group.ts
import type { FineTypeDto } from './fineTypes';
import type { UserDto } from './user';

export type CreateGroupDto = {
	name: string;
};

export type GroupDto = {
	groupId: number;
	name: string;
};

export type GroupDetailDto = {
	groupId: number;
	name: string;
	fineTypes: FineTypeDto[];
	admins: UserDto[];
	members: UserDto[];
	pendingMembers: UserDto[];
};
